### PR TITLE
[release-4.10] OCPBUGS-16206: Fix stale SNAT entries for completed pods

### PR DESCRIPTION
**Pull request by pperiyasamy**
Manual cherry-pick of 56c68141134451cff2af24a4ca62ad2742ddc421.

Resolved Conflicts:
- `go-controller/pkg/ovn/egressip_test.go` [1](https://github.com/openshift/ovn-kubernetes/pull/1758/files#diff-56e7d797e8b9524f28eb9baa440e18600a173007ec0850958b3e1c30da4bb4baR4994-R4998).
- `go-controller/pkg/ovn/egressip_test.go` [2](https://github.com/openshift/ovn-kubernetes/pull/1758/files#diff-56e7d797e8b9524f28eb9baa440e18600a173007ec0850958b3e1c30da4bb4baR5154-R5158).
- `go-controller/pkg/ovn/egressip_test.go` [3](https://github.com/openshift/ovn-kubernetes/pull/1758/files#diff-56e7d797e8b9524f28eb9baa440e18600a173007ec0850958b3e1c30da4bb4baR5242-R5251).
- `go-controller/pkg/ovn/egressip_test.go` [4](https://github.com/openshift/ovn-kubernetes/pull/1758/files#diff-56e7d797e8b9524f28eb9baa440e18600a173007ec0850958b3e1c30da4bb4baR5263).

Signed-off-by: Periyasamy Palanisamy [pepalani@redhat.com](mailto:pepalani@redhat.com)
(cherry picked from commit 7b88e0acbfa7041ad056d32f7b0169565b44f874) 
(cherry picked from commit cd99b41e441a917e51f983f69d34578167293395) 
(cherry picked from commit c3fbfed12bca5f55682ecb1b5461a4b91c5d7649) 
(cherry picked from commit 56c68141134451cff2af24a4ca62ad2742ddc421)

### CODE REVIEW

**Review by flavio-fernandes — APPROVED** [review]
I'm hoping we will no longer need to be backporting to 4.10.
But assuming we do...
/lgtm

**Comment by jcaamano** [comment]
@pperiyasamy do you know why we needed the no reroute policy added compared to the 4.11 changes?

File: go-controller/pkg/ovn/egressip_test.go [hunk 1]
```diff
@@ -4890,6 +4890,396 @@ var _ = ginkgo.Describe("OVN master EgressIP Operations", func() {
...
+				_, err = fakeOvn.fakeClient.KubeClient.CoreV1().Pods(egressPod.Namespace).Update(context.TODO(), &egressPod, metav1.UpdateOptions{})
+				gomega.Expect(err).ToNot(gomega.HaveOccurred())
+				// Wait for pod to get moved into succeeded state.
+				gomega.Eventually(func() v1.PodPhase {
+					egressPod1, _ := fakeOvn.watcher.GetPod(egressPod.Namespace, egressPod.Name)
+					return egressPod1.Status.Phase
+				}, 5).Should(gomega.Equal(kapi.PodSucceeded))
+
+				node1GR.Nat = []string{}
+				podNoReRoutePolicy := &nbdb.LogicalRouterPolicy{
```
Comment (jcaamano) [inline go-controller/pkg/ovn/egressip_test.go:5242]: We looked this up and this is the egress ip policy that does not get garbage collected.

I think the variable name should be `podReRoutePolicy`

Comment (pperiyasamy) [reply go-controller/pkg/ovn/egressip_test.go]: done, changed it with `podReRoutePolicy.Nexthops = nil`.
Thanks @jcaamano for your time in debugging this.

**Comment by flavio-fernandes** [comment]
@dcbw @trozet  @jcaamano may we 
`/override ci/prow/e2e-metal-ipi-ovn-dualstack`
?

**Comment by dcbw** [comment]
/override ci/prow/e2e-metal-ipi-ovn-dualstack

`: [sig-network] [Feature:IPv6DualStack] should have ipv4 and ipv6 internal node ip [Suite:openshift/conformance/parallel] [Suite:k8s]` failure is not related to this test; it's infrastructure/metal setup.

**Pull request merged**